system/modules/core/classes/FrontendTemplate.php

```diff
@@ -213,7 +213,7 @@ protected function addToCache()
 			// If the request string is empty, use a special cache tag which considers the page language
 			if (\Environment::get('request') == '' || \Environment::get('request') == 'index.php')
 			{
-				$strCacheKey = \Environment::get('host') . '/empty.' . $objPage->language;
+				$strCacheKey = ($objPage->domain ?: '*') . '/empty.' . $objPage->language;
```

Comment:
No, `$objPage->dns` is only set for root pages. For other pages this information is saves in `->domain`. See [system/modules/core/models/PageModel.php#L868](https://github.com/contao/core/blob/3.5.25/system/modules/core/models/PageModel.php#L868)